test/35fa71a030ca.c

```diff
@@ -28,6 +28,7 @@
 #include "helpers.h"
 #include "../src/syscall.h"
 
+#ifdef CONFIG_USE_SANITIZER
```

Comment:
Like I've mentioned many times (sorry!), these changes should be BEFORE the commit adding sanitizer support. Not a huge deal and if that's the only thing left, then I'm inclined to just roll with it in the spirit of moving this forward.

BUT... the main commit adding sanitizer support should be JUST adding sanitizer support to the core, not fiddling with any tests at all. So ideally you'd yank the test changes out of this:

1) The ones touching existing tests, should go BEFORE the commit adding the sanitizer support. It prepares them for when the sanitizer code exists, and will obviously be a no-op before that
2) The ones adding new sanitizer tests, ideally would go AFTER the commit adding sanitizer support. This stands alone imho, as it's adding tests that test the sanitizer code.

Reply:
Agree, I'll write something like that. Usually people just send smaller things and it's just about the commit message, but I can cover both.

Reply:
Can you take a look at:

https://github.com/axboe/liburing/blob/master/CONTRIBUTING.md

when you have time? Would appreciate your feedback, especially if there are things I've missed. Consider it a first cut at trying to provide the rules for liburing commits, and reasonings behind it.

Reply:
> Can you take a look at:
> 
> https://github.com/axboe/liburing/blob/master/CONTRIBUTING.md
> 
> when you have time? Would appreciate your feedback, especially if there are things I've missed. Consider it a first cut at trying to provide the rules for liburing commits, and reasonings behind it.

Here's a couple of things that I think would be valuable to add:

* A list of style examples (space after if/while/for, braces on next line for functions, same line for if statements, no braces for 1-statement ifs/for's, where to put the *, variable definitions on top, etc)
* Explicitly mentioning that there is a preference for separating tests from code changes in separate commits

Reply:
Oh, and you could point towards an example PR (this one maybe) for how reviews go and how it should end up as.

Reply:
> Explicitly mentioning that there is a preference for separating tests from code changes in separate commits

To me, this should not be necessary, as it falls under the "commit should do 1 thing, and 1 thing only" rule.

For coding style, I think I'll just mention that any changes should follow the style of the code around it.

Reply:
Fair point. Good additions otherwise!

Reply:
> Here's a couple of things that I think would be valuable to add:
> 
> * A list of style examples (space after if/while/for, braces on next line for functions, same line for if statements, no braces for 1-statement ifs/for's, where to put the *, variable definitions on top, etc)

I believe you can follow the Linux kernel coding style, as liburing is mainly written by Linux kernel developers.

See the list of style examples here: 
https://www.kernel.org/doc/html/latest/process/coding-style.html